---
title: Day 36 - Fixing the comments API ðŸ”¨
date: 2022-01-17
description: getting Django to behave properly...
tags: Django, Docker
---

# {{ $page.title }}

This really must be fixed. 

## What is the current status? ðŸ¤”

It has been a while since I last took on any Django, so one must reassess the situation. The live container seems devoid of the routing issue that would point to both the front-end and the API, which is nice, but every request is still met with a pesky 500. 

Even running the local version will incur in some undesirable problems: 400 error on what looks like a valid route. In fact, **every** endpoint is showing 400. Is this project doomed beyond repair? This was actually quite easy to sort out: I have a ``IS_ON_DEV`` boolean that wasn't detecting this particular computer as a valid development. Easy hurdle. Plus, I had it deployed without even passing Django through a `gunicorn` instance. That is not best practice, but is fixed now.

## Finding the error! ðŸ§¶

The query consistently fails with the following error:

``` python
django.db.utils.OperationalError: no such table: api_comment
```

Weird. This was usually solved, on development, with ``python3 manage.py migrate --sync-db``
